usr/share/rear/lib/layout-functions.sh

```diff
@@ -611,3 +611,207 @@ retry_command ()
+    device_link="$1"
+
+    # Exit with Error if no argument is provided to UdevSymlinkName
+    test $device_link || Error "Empty string passed to UdevQueryName()"
+
+    # be careful udevinfo is old, now we have udevadm
+    # udevinfo -r -q name -n /dev/disk/by-id/scsi-360060e8015268c000001268c000065c0-part4
+    # udevadm info --query=name --name /dev/disk/by-id/dm-name-vg_fedora-lv_root
+    type -p udevinfo >/dev/null && UdevQueryName="udevinfo -r -q name -n"
+    type -p udevadm >/dev/null && UdevQueryName="udevadm info --query=name --name"
+
+    if test -z "$UdevQueryName" ; then
+        LogPrint "Could not find udevinfo nor udevadm. UdevQueryName($device_link) failed."
+        return 1
+    fi
+
+    $UdevQueryName $device_link
+}
+
+
+### apply-mappings function
+#
+# Functions used in apply-mappings() function
+add_replacement() {
+    echo "$1 _REAR${replaced_count}_" >> "$replacement_file"
+    let replaced_count++
+}
+#
+# Functions used in apply-mappings() function
+has_replacement() {
+    if grep -q "^$1 " "$replacement_file" ; then
+        return 0
+    else
+        return 1
+    fi
+}
+#
+# Functions used in apply-mappings() function
+get_replacement() {
+    local item replacement junk
+    read item replacement junk < <(grep "^$1 " $replacement_file)
+    echo "$replacement"
+}
+#
+# apply-mappings function migrate disk device reference from an old system and
+# replace them with new one (from current system).
+# the relationship between OLD and NEW device is provided by $MAPPING_FILE
+# (usually disk_mappings file in $VAR_DIR).
+apply-mappings() {
+
+    # apply-mappings need one argument (file which contains disk device to migrate).
+    if [ -z "$1" ] ; then
+        LogError "apply-mappings function called without argument (file_to_migrate)"
+    else
+        file_to_migrate="$1"
+    fi
+
+    if [ -z "$MIGRATION_MODE" ] ; then
+        return 0
+    fi
+
+    # We temporarily map all devices in the mapping to new names _REAR[0-9]+_
+    replaced_count=0
+    replacement_file="$TMP_DIR/replacement_file"
+    : > "$replacement_file"
+
+    # Generate replacements.
+    while read source target junk ; do
+        if ! has_replacement "$source" ; then
+            add_replacement "$source"
+        fi
+
+        if ! has_replacement "$target" ; then
+            add_replacement "$target"
+        fi
+    done < "$MAPPING_FILE"
+
+    # Replace all originals with their replacements.
+    while read original replacement junk ; do
+        # Replace partitions (we normalize cciss/c0d0p1 to _REAR5_1)
+        part_base="$original"
+        case "$original" in
+            *mmcblk[0-9]*|*nvme[0-9]*n[1-9]*|*rd[/!]c[0-9]*d[0-9]*|*cciss[/!]c[0-9]*d[0-9]*|*ida[/!]c[0-9]*d[0-9]*|*amiraid[/!]ar[0-9]*|*emd[/!][0-9]*|*ataraid[/!]d[0-9]*|*carmel[/!][0-9]*)
+                part_base="${original}p" # append p between main device and partitions
+                ;;
+            *mapper[/!]*)
+                case $OS_VENDOR in
```

Comment:
I am not sure about this after @jsmeix changed the way it works, but I'd have expected here `$OS_MASTER_VENDOR`

Reply:
As far as I understood at the time of
https://github.com/rear/rear/commit/5b35338f5eb1a2a76621443fbf2055a7d559724e
and
https://github.com/rear/rear/commit/9c42a6a3807ca207e721d07ae797dbd9b6188bf6
I did not changed the way it works.
I only set for SUSE those variables in a way
as I saw it was already done for other distributions.
I.e. I tried to be compatible.

Reply:
Then indeed it might be better to use `$OS_MASTER_VENDOR` because it will catch you all Debian derivatives under Debian etc.

Reply:
@jsmeix can you give me more information about this ? 
- difference between `$OS_VENDOR` and `$OS_MASTER_VENDOR`
- Is there a variable which could act as a "family" name like FEDORA=(redhat,centos,fedora) DEBIAN=(ubuntu,debian).

Reply:
@jsmeix Ok I got my answer from `config-functions.sh`